src/IO/HDF5/HDF5IOHandler.cpp

```diff
@@ -1465,9 +1465,9 @@ HDF5IOHandlerImpl::readAttribute(Writable* writable,
                 char* m2 = H5Tget_member_name(attr_type, 2);
                 if(strcmp("x", m0) != 0 || strcmp("y", m1) != 0 || strcmp("z", m2) != 0)
```

Comment:
I tested this and this fixed the issue we saw.
But looking at the difference between `free` and `H5free_memory` I wonder if, as a follow-up, we should consider that `` might return a `nullptr`:
https://support.hdfgroup.org/HDF5/doc/RM/RM_H5T.html#Datatype-GetMemberName
```
Returns:
    Returns a valid pointer to a string allocated with malloc() if successful; otherwise returns NULL.
```

The behavior of using a `nullptr` in `strcmp` seems to be implementation-defined (we might be lucky atm.):
https://stackoverflow.com/questions/21865041/in-c-what-exactly-happens-when-you-pass-a-null-pointer-to-strcmp